pkg/daemon/server/service/pipeline_watermark_query.go

```diff
@@ -142,18 +144,26 @@ func (ps *pipelineMetadataQuery) GetPipelineWatermarks(ctx context.Context, requ
 	i := 0
 	for k, vertexFetchers := range ps.watermarkFetchers {
 		var latestWatermark = int64(-1)
+		var latestpodWatermarks []processor.Watermark
 		for _, fetcher := range vertexFetchers {
 			watermark := fetcher.GetHeadWatermark().UnixMilli()
+			podWatermark := fetcher.GetPodWatermarks()
```

Comment:
My understanding of doing this is:

1. Update the return from `watermark` to `watermarks` (array);
2. Update `GetVertexWatermarkFetchers`, to update the `key` of the map from `vertex` to `vertex/replica` - for map vertex, the `replica` could be always `0`.
3. When querying wartermarks, use corresponding fetcher(s) from the map to do it.

Reply:
updated as mentioned

Reply:
will be moving the watermark changes to a new pr as discussed